### PR TITLE
MINOR: Add try-with-resources around KafkaStreams instances

### PR DESCRIPTION
**Pull request by bbejeck**

### CODE REVIEW

File: kafka-streams/src/main/java/io/confluent/developer/basic/solution/BasicStreams.java [hunk 1]
```diff
@@ -36,10 +36,9 @@ public static void main(String[] args) throws IOException {
...
 
-        KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), streamsProps);
-        
-        TopicLoader.runProducer();
-
-        kafkaStreams.start();
+       try(KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), streamsProps)) {
+           TopicLoader.runProducer();
+           kafkaStreams.start();
+       }
```
Comment (davetroiano) [inline kafka-streams/src/main/java/io/confluent/developer/basic/solution/BasicStreams.java:42]: indent one more space

**Review by davetroiano — COMMENTED** [review]
not introduced by this PR I see some newline inconsistency across these examples at the end of main methods. maybe worth a sweep since you're in the area. I like no newline before method-ending brace, newline before class-ending brace, and newline to end file, but really anything is fine as long as we're consistent:
```
        ...

        try(KafkaStreams kafkaStreams = new KafkaStreams(builder.build(), streamsProps)) {
            // Set an UncaughtExceptionHandler on Kafka Streams HINT: It should be type of StreamsUncaughtExceptionHandler
            // kafkaStreams.set...
            TopicLoader.runProducer();
            kafkaStreams.start();
        }
    }

}

```
feel free to open a separate issue if the whitespace inconsistency is more widespread

**Comment by bbejeck** [comment]
@davetroiano updated per comments

**Review by davetroiano — APPROVED** [review]
LGTM

**Pull request merged**